spec/inverted-index-spec.js

```diff
@@ -1,74 +1,96 @@
-  const InvertedIndex = new IndexKlass(InvertedIndexhelper);
+  const InvertedIndex = new IndexClass(InvertedIndexhelper);
   const file1 = book;
-  const emptyFile = [];
-  const notJson = 'books.js';
-  InvertedIndex.files.books = JSON.parse(book);
-  InvertedIndex.createIndex('books');
+  const file2 = empty;
+  const file3 = invalidFile;
+
   afterEach(() => {
     InvertedIndex.indexTable = {};
   });
-
-
+  
   it('should be truthy for the instance of the class', () => {
-    expect(InvertedIndex instanceof IndexKlass).toBeTruthy();
+    const indexInstance = InvertedIndex;
+    expect(InvertedIndex instanceof IndexClass).toBeTruthy();
   });
-
   it('should return zero for the length of the indexes', () => {
-    expect(Object.keys(InvertedIndex.getIndex).length).toBe(0);
+    const indexInstance = InvertedIndex;
+    expect(Object.keys(InvertedIndex.indexTable).length).toBe(0);
   });
 
   describe('Read Book Data', () => {
-    it('return false if the content of the file is empty', () => {
-      expect(InvertedIndexhelper.readBookData(emptyFile)).toBe(false);
+    const notEmpty = file2;
+    it('throws error if file is empty', () => {
+      expect(() => InvertedIndexhelper.isValidFile(file2)).toThrow(new Error("File invalid"));
     });
-    it('returns true if the content of the file is a valid JSON array', () => {
-      const isValid = InvertedIndexhelper.readBookData(file1);
-      expect(InvertedIndexhelper.readBookData(isValid)).toBe(false);
+    // should return boolean - test for empty array
+    it('returns an array if the content of the file is a valid JSON array', () => {
+      const file = InvertedIndexhelper.isValidFile(file1);
+      expect(file).toBeTruthy();
     });
     it('returns false if the content of the file is not a valid JSON array', () => {
-      expect(InvertedIndexhelper.readBookData(notJson)).toBe(false);
-    });
-    it('returns the content of the file when an index has been created', () => {
-      const createIndex = InvertedIndexhelper.readBookData(file1);
-      expect(createIndex).toBeDefined();
+      expect(() => InvertedIndexhelper.isValidFile(file3)).toThrow(new Error("File invalid"));
     });
-    it('returns false if the file has been uploaded before', () => {
-      const createIndex = InvertedIndex.createIndex('book');
-      expect(createIndex).toBe(false);
+    it('does not overwrite previously created indices', () => {
+      const validBook = JSON.parse(book);
+      InvertedIndex.createIndex('bk', validBook);
+      InvertedIndex.createIndex('bk2', validBook);
+      expect(InvertedIndex.indexTable.hasOwnProperty('bk')).toBe(true);
     });
   });
 
   describe('Populate Index', () => {
-    it('ensures that index is created', () => {
-      expect(InvertedIndex.createIndex('bk')).toBe(false);
+    it('returns false if file content is not passed', () => {
+      const fileContent = null;
+      expect(InvertedIndex.createIndex('bk', fileContent)).toBe(false);
```

Comment:
### Critical

`fileContent` was never defined, if this was the intention, it'll make more sense to use either `undefined` or and empty string `''`